Stockwatch/Stockwatch.Businesses/StockDataService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Stockwatch.Model;
+
+namespace Stockwatch.Business
+{
+    public interface IStockDataservice {
+        void AddStock(StockData StockData);
+        void UpdateStock(StockData StockData);
+        void RemoveStock(StockData StockData);
+        StockData FetchName(string Name);
+        StockData Fetchstock(int id);
+        List<StockData> GetAll();
+    }
+    public class StockDataservice: IStockDataservice
+    {
+        private readonly StockwatchDbContext _context;
+
+        public StockDataservice(StockwatchDbContext context) { _context = context; }
+
+        public void AddStock(StockData StockData)
+        {
+            _context.StockDatas.Add(StockData);
+            _context.SaveChanges();
+        }
+
+        public void UpdateStock(StockData StockData)
+        {
+            _context.Update(StockData);
+            _context.SaveChanges();
+        }
+
+        public StockData FetchName(string Name)
+        {
+            return GetAll().Find(x => x.StockSymbol.SymbolName == Name);
```

Comment:
Using GetAll() here is a wrong way of doing this method, check what happens when we call ToList in EF

Reply:
Changed. Will be called from context and checked